packages/core/src/processing/generator/generated_data.rs

```diff
@@ -78,11 +80,34 @@ impl GeneratedData {
         };
 
         // write header and records
-        for r in synthetic_data.iter() {
-            match wtr.write_record(r.iter().map(|v| v.as_str())) {
+        if long_form {
```

Comment:
Maybe the content of the if and else blocks could be encapsulated in its own function, something like:

`_write_synthethic_data_long_format` and `_write_synthethic_data_raw_format`

Both could be annotated with `#[inline]`, which would give a hint to the compiler to generate the code without the function itself, avoiding the extra function call in runtime.